src/main/java/nextstep/subway/line/domain/SectionLineUp.java

```diff
@@ -98,79 +106,73 @@ public List<Station> getStations() {
+                .findFirst()
+                .map(it -> findDownStation(it.getDownStation()))
+                .orElse(station);
     }
 
-    private Station findDownStation() {
-        Station upStation = sections.get(FIRST_INDEX).getDownStation();
-        while (upStation != null) {
-            Station finalUpStation = upStation;
-            Optional<Section> nextLineStation = sections.stream()
-                    .filter(it -> it.getUpStation().equals(finalUpStation))
-                    .findFirst();
-            if (!nextLineStation.isPresent()) {
-                break;
-            }
-            upStation = nextLineStation.get().getDownStation();
+    public void remove(Station station) {
+        validMinSection();
+        validNotHasStation(station);
+
+        if (isInternalSection(station)) {
+            // 이미 upSection과 downSection이 존재하는 것이 검증 되었기에 Optional 값에 대한 별도 체크를 하지 않는다
+            Section upSection = this.sections.stream().filter(it -> it.isSameDownSection(station)).findFirst().get();
+            Section downSection = this.sections.stream().filter(it -> it.isSameUpSection(station)).findFirst().get();
+            sections.add(Section.mergeSection(upSection, downSection));
+            sections.remove(upSection);
+            sections.remove(downSection);
+            return;
         }
+        removeEndSection(station);
+    }
 
-        return upStation;
+    private void removeEndSection(Station station) {
+        sections.stream().filter(it -> it.isSameUpSection(station) || it.isSameDownSection(station))
+                .findFirst()
+                .ifPresent(it -> sections.remove(it));
     }
 
-    public void remove(Station station) {
-        if (this.sections.size() <= 1) {
-            throw new IllegalStateException("구간 삭제는 구간이 2개 이상일 경우 가능합니다");
-        }
+    private boolean isInternalSection(Station station) {
+        return sections.stream().anyMatch(it -> it.isSameDownSection(station)) &&
+                sections.stream().anyMatch(it -> it.isSameUpSection(station));
+    }
 
-        Optional<Section> upLineStation = this.sections.stream()
-                .filter(it -> it.getUpStation().equals(station))
-                .findFirst();
-        Optional<Section> downLineStation = this.sections.stream()
-                .filter(it -> it.getDownStation().equals(station))
-                .findFirst();
-
-        if (upLineStation.isPresent() && downLineStation.isPresent()) {
-            Station newUpStation = downLineStation.get().getUpStation();
-            Station newDownStation = upLineStation.get().getDownStation();
-            Distance newDistance = upLineStation.get().getDistance().plus(downLineStation.get().getDistance());
-            this.sections.add(
-                    new Section(sections.get(FIRST_INDEX).getLine(), newUpStation, newDownStation, newDistance));
+    private void validNotHasStation(Station station) {
+        if (!this.getStations().contains(station)) {
```

Comment:
```suggestion
        if (!this.isStationContains(station)) {
```

좀 더 명확한 메시지를 만들어 보는것도 좋겠네요!

Reply:
정보를 물어보는 구조에서 메세지를 전달하는 구조로 분리가 가능하겠네요 !
피드백 내용 반영하였습니다 ! 감사합니다 👍